docs/runtimes-building.md

```diff
@@ -0,0 +1,146 @@
+# license agreements.  See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.  The ASF licenses this file to you
+# under the Apache License, Version 2.0 (the # "License"); you may not use this
+# file except in compliance with the License.  You may obtain a copy of the License
+# at:
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+-->
+
+# Working with Runtimes locally
+
+This document is designed to be applied to the various
+`incubator-openwhisk-runtimes-*` repositories as they are converted to
+multi-architecture build. The multi-architecture build supports building Docker
+multi-architecture manifest lists across (currently) amd64, Power 64
+(little-endian), and System Z (s390x) architectures.  The framework is designed
+to extend to other chips, including the Arm64 chipset.
+
+You can tell that a runtime has been converted to a multi-architecture build
+by referring to its root `README.md` file.
+
+## Reading this document
+
+For this document, we will assume an environment variable `ACTION_NAME` has
+been set to the directory name of an action in the runtime repository on which
+work is being done (i.e. set to `nodejs6Action` while working in
+`incubator-openwhisk-runtime-nodejs`).
+
+## Building a single-architecture Docker image locally
+
+To build a single-architecture image locally on a 64-bit x86 machine, execute
+
+```
+./gradlew core:$ACTION_NAME:dockerBuildImage -Pdocker_local_json='{"amd64":null}'
+```
+
+To check on what images will be produced, look in the `settings.json` file in
+the runtime repository (look for `gradle.ext.dockerBuildProjects =`).  The
+default image prefix is currently 'whisk'.
+
+## Building a multi-architecture Docker image
+
+Multi-architecture builds work be repeating build steps on independent docker
+instances for each architecture. To ensure security, it's necessary to set up
+the docker instances to use TLS and to have the TLS client certificates (
+`ca.pem`, `cert.pem` and `key.pem`) available in some directory on the local
+machine.  For details on setting up docker with TLS see [the official docker
+documentation](https://docs.docker.com/engine/security/https/).
+
+To build a multi-architecture image requires providing more sophisticated
+JSON defining the build docker environment for each architecture.  While this
+could be done on the command line (as above), it is generally easier to create
+a `docker-local.json` file in the root directory of the runtime.  (Don't
+worry, it's `.gitignore`d and won't find its way into the open.)
+
+Here's a (sanitized) example of a docker-local.json file:
+
+```
+{
+    "amd64": null,
```

Comment:
i find the null strange - but now i understand.

Reply:
Yeah, null means undefined so fall back on default.  There may be a better way of expressing that sentiment, but this works will with the code I have.